File: app/src/main/java/com/example/mapapp/data/Item.java
package com.example.mapapp.data;

import org.json.JSONObject;

/**
 * Created by ext-jurkeant on 9/28/2016.
 */

public class Item implements JSONPopulator {
    private Condition condition;
    private double latitude;
    private double longtitude;

    @Override
    public void populate(JSONObject data) {
        condition = new Condition();
        condition.populate(data.optJSONObject("condition"));
        latitude = data.optDouble("latitude");
        longtitude = data.optDouble("longtitude");
    }

    public Condition getCondition() {
        return condition;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongtitude() {
        return longtitude;
    }
}

File: app/src/main/java/com/example/mapapp/data/Item.kt
package com.example.mapapp.data

import org.json.JSONObject

/**
 * Created by ext-jurkeant on 9/28/2016.
 */

class Item : JSONPopulator {
    var condition: Condition? = null
        private set
    var latitude: Double = 0.toDouble()
        private set
    var longtitude: Double = 0.toDouble()
        private set

    override fun populate(data: JSONObject) {
        condition = Condition()
        condition!!.populate(data.optJSONObject("condition"))
        latitude = data.optDouble("latitude")
        longtitude = data.optDouble("longtitude")
    }

}
